Nasal/engine-fire.nas

```diff
@@ -0,0 +1,119 @@
+# Engine Fire Originally By Gabriel Hernandez (YV3399)
+# Edited for the 737-800 by Jonathan Redpath
+# Copyright (c) 2018 Jonathan Redpath
+# GNU GPL 2.0
+
+fire_init = 0;
+
+var fire_mgmt = {
+    init : func { 
+        me.UPDATE_INTERVAL = 1; 
+        me.loopid = 0;
+        me.reset(); 
+        
+        for (var i = 0; i < 3; i+= 1) {
+            createFireListener(i);
+            setprop("/controls/fires/fire[" ~ i ~ "]/extinguish-handle-armed", 0);
+            setprop("/controls/fires/fire[" ~ i ~ "]/extinguish-handle-pos", 0);
+            setprop("/controls/fires/fire[" ~ i ~ "]/burn-time", 0);
+            setprop("/controls/fires/fire[" ~ i ~ "]/phase", "no-fire");
+            setprop("/sim/sound/explode"~i, 0);
+        }
+        
+        createFireLampListener(0, "engine1");
+        createFireLampListener(1, "engine2");
+        createFireLampListener(2, "apu");
+    }, 
+    update : func {
+    
```

Comment:
This should be changed to use listeners on /controls/fires/fire[n]/on-fire - but I'll do that as a separate commit, no need to complicate this commit.

Reply:
So in essence, exactly this, but with update() only being run if a fire exists?

Reply:
Precisely.